lib/active_record_upsert/active_record/persistence.rb

```diff
@@ -15,7 +15,7 @@ def upsert!(attributes: nil, arel_condition: nil, validate: true)
             _upsert_record(attributes.map(&:to_s).uniq, arel_condition)
           }
         }
-        assign_attributes(values.first.to_h)
+        assign_attributes(values.first.to_h.slice(*self.attributes.keys))
```

Comment:
Is there a chance to test this somehow (in specs)?

Reply:
As noted above... Not sure how I would test this. The error requires the underlying schema to change after the model has been loaded...